### PR TITLE
s22-4pm-4: Vr backend degradation rate field added

### LINKED ISSUE #17: BackEnd Degradation Rate
To the Backend create update commons APIs, and the list API add the field "degradation rate", a decimal number treated as a percentage. (It may be any positive value or zero, including values > 100). Negative values should be disallowed.

-[ ] degradation field exists in backend

### LINKED ISSUE #18: Entity Degradate Rate Field
To the commons database table, add a field degradation rate. It should be a decimal number (i.e. a Double, not a Long).

This means altering the entity only (for this particular issue.)

### PR DESCRIPTION
**Pull request by vmreed1**
# Overview

I added backend code for a double variable called degradationRate. This variable takes in a decimal value (must be 0 or larger- no negatives).It throws an error if a negative rate is entered. I added code in the controller and entity for this variable. Appears on backend when run on localhosts.  (pull requests linked to issues on Kanban Board - this pull request covers two issues: adding Backend Degradation variable to Entity  and adding Degradation field to CommonsController) 
Screen Shot 2022-05-27 at 4 01 38 PM
Screen Shot 2022-05-27 at 4 01 20 PM
Screen Shot 2022-05-27 at 4 00 01 PM
Screen Shot 2022-05-27 at 3 59 47 PM

### CODE REVIEW

**Review by aripo99 — DISMISSED** [review]
LGTM!

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java [hunk 1]
```diff
@@ -64,48 +66,138 @@ public class CommonsControllerTests extends ControllerTestCase {
...
+        .perform(post("/api/commons/new").with(csrf())
+            .contentType(MediaType.APPLICATION_JSON)
+            .characterEncoding("utf-8")
+            .content(requestBody))
+            .andExpect(status().isBadRequest()).andReturn();
+
+        Optional<IllegalArgumentException> someException = Optional
+        .ofNullable((IllegalArgumentException) response.getResolvedException());
+
+    //someException.ifPresent((se) -> assertNotNull(se));
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:194]: We should remove these commented out lines of code.

**Review by pconrad — CHANGES_REQUESTED** [review]
There are some commented out lines of code that should be removed.

Otherwise LGTM... once that's done and it's tested on QA, this one looks good to merge.

**Comment by zoeyz5** [comment]
LGTM

**Review by htn5 — APPROVED** [review]
LGTM

**Review by zoeyz5 — APPROVED** [review]

**Review by zoeyz5 — APPROVED** [review]
LGTM

**Comment by kheff16** [comment]
@vmreed1 looks like there might be a 500 on get /all for this pr. I went back to test on the main deployment to confirm that this is novel, and it is. Here's a screenshot of the dev tools. 
image

**Review by pconrad — APPROVED** [review]
LGTM!

**Comment by WangHaoranRobin** [comment]
Could not replicate. Maybe try "mvn clean"?

**Pull request merged**